compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2278,6 +2284,10 @@ fn add_order_independent_options(
     out_filename: &Path,
     tmpdir: &Path,
 ) {
+    if flavor.uses_clang() && sess.target.linker_flavor != sess.host.linker_flavor {
+        cmd.arg(format!("--target={}", sess.target.llvm_target));
```

Comment:
Is this condition sufficient for the general case of adding `--target` when cross-compiling? The linker flavor doesn't contain any information about the target triple (unless I'm missing something), so I don't see how this can reliably detect cross-compiling.

Reply:
From what I know `--target` only affects which linker is looked up (`ld.lld`, or `ld64.lld`, etc) if clang is used for linking only.
So if the required flavor is the same as host, then clang without `--target` will already use the right linker.

On the other hand, there are some issues with crt object versions on macOS if `--target` is specified (fixed by https://github.com/rust-lang/rust/pull/101792), maybe target takes priority over some relevant environment variables.
So always passing `--target` leads to issues.

We'll likely need to tweak this logic based on reported issues.